### ISSUE TITLE
NoMethodError when statement ends on '.' 

### ISSUE THREAD

**Issue by tomv564** [castwide/solargraph#89]
This might be a recent regression with source_chainer? Running 0.28.1 on os-x.

When adding the following dot (in Sublime with LSP package), the immediate completion request fails. Hover also fails with the same error.

```
msg = 'Hello World'
msg.
```

Server logs:
```
LSP:  --> textDocument/didChange
LSP:  --> textDocument/completion
server: undefined method `include?' for nil:NilClass
server: /Library/Ruby/Gems/2.3.0/gems/solargraph-0.28.1/lib/solargraph/source/source_chainer.rb:45:in `block in chain'
server: /Library/Ruby/Gems/2.3.0/gems/solargraph-0.28.1/lib/solargraph/source/source_chainer.rb:45:in `any?'
server: /Library/Ruby/Gems/2.3.0/gems/solargraph-0.28.1/lib/solargraph/source/source_chainer.rb:45:in `chain'
server: /Library/Ruby/Gems/2.3.0/gems/solargraph-0.28.1/lib/solargraph/source/source_chainer.rb:19:in `chain'
server: /Library/Ruby/Gems/2.3.0/gems/solargraph-0.28.1/lib/solargraph/source/cursor.rb:75:in `chain'
server: /Library/Ruby/Gems/2.3.0/gems/solargraph-0.28.1/lib/solargraph/source_map/clip.rb:24:in `complete'
server: /Library/Ruby/Gems/2.3.0/gems/solargraph-0.28.1/lib/solargraph/library.rb:144:in `completions_at'
server: /Library/Ruby/Gems/2.3.0/gems/solargraph-0.28.1/lib/solargraph/language_server/host.rb:352:in `completions_at'
```

**Comment by castwide** [comment]
It looks like the source received a null range while checking for updates with syntax errors from the `textDocument/didChange` message. The specification allows for null ranges, so the chainer will have to check for them.

I'll add a fix to the next patch release.

**Comment by tomv564** [comment]
Thanks! I'd like to send partial changes in `didChange` but did not find a performant way to do it in Sublime Text yet (without full-text diffing).

Please close at your convenience :)

**Comment by castwide** [comment]
The error is fixed in version 0.28.2.

Incremental changes will still yield better results for certain completion requests due to the way that syntax errors get repaired in updates. Improving repairs for full-text changes is on the roadmap.
